### PR TITLE
Add support for transferring non-bagged zipfiles

### PR DESCRIPTION
**Pull request by helenst**
Add an option to the transfer type dropdown allowing a plain zipfile to be selected.

The MCPClient script for extracting bagged zips has been adapted and renamed so that it is used for both bagged and unbagged zips, since the code is so similar. A flag is provided to distinguish whether the file is expected to be a bag.

Connected to: https://github.com/archivematica/Issues/issues/682
Docs: https://github.com/artefactual/archivematica-docs/pull/353
Sampledata: https://github.com/artefactual/archivematica-sampledata/pull/69

### CODE REVIEW

**Comment by sromkey** [comment]
Hey @artefactual/amcr group, I think this is ready for CR based on discussion with Helen on the issue.

File: src/MCPClient/lib/clientScripts/extract_zipped_transfer.py [hunk 1]
```diff
@@ -69,18 +69,34 @@ def extract(job, target, destinationDirectory):
...
+
                 destinationDirectory = os.path.join(processingDirectory, basename)
 
-                destinationDirectory = get_extract_dir_name(destinationDirectory)
+                # trim off '.tar' if present (os.path.basename doesn't deal well with '.tar.gz')
+                try:
+                    tar_extension_position = destinationDirectory.rindex(".tar")
+                    destinationDirectory = destinationDirectory[:tar_extension_position]
+                except ValueError:
+                    pass
```
Comment (ross-spencer) [inline src/MCPClient/lib/clientScripts/extract_zipped_transfer.py]: This code has moved significantly thanks to Helen's work. The work done here for tar files is now done outside of this function. I have created a reference rebase [here](https://github.com/artefactual/archivematica/blob/dev/issue-682-support-zipped-transfers/src/MCPClient/lib/clientScripts/extract_zipped_transfer.py). My suggestion is to review that copy, compare it with the one currently being run at Wellcome, and then submit it back against this PR. As far as I can see it wouldn't require any code changes once rebased. Other  suggestions welcome.

File: src/MCPServer/lib/assets/workflow.json [hunk 2]
```diff
@@ -4306,6 +4331,35 @@
...
+            "exit_codes": {},
+            "fallback_job_status": "Failed",
+            "fallback_link_id": null,
+            "group": {
+                "en": "Approve transfer",
+                "es": "Aprobar transferencia",
+                "fr": "Approuver le transfert",
+                "ja": "転送を承認",
+                "pt_BR": "Confirmar transferência",
+                "sv": "Godkänn överföring"
```
Comment (ross-spencer) [inline src/MCPServer/lib/assets/workflow.json:4360]: Just noting, these changes all look sensible. The new transfer type follows a couple of setup processes before eventually following the standard transfer path. 

**Transfer starts and is approved**

image

**Transfer is extracted**

image

**Transfer follows the standard path**

image

File: src/MCPServer/lib/assets/workflow.json [hunk 3]
```diff
@@ -12158,6 +12247,12 @@
             "path": "/activeTransfers/standardTransfer",
             "unit_type": "Transfer"
         },
+        {
+            "chain_id": "a10b0088-b038-4ea2-91cc-b7761f07fe5a",
+            "only_dirs": false,
+            "path": "/activeTransfers/zippedDirectory",
```
Comment (ross-spencer) [inline src/MCPServer/lib/assets/workflow.json:12253]: This directory is created as anticipated:
```
activeTransfers# tree
.
├── baggitDirectory
├── baggitZippedDirectory
├── dataverseTransfer
├── Dspace
├── maildir
├── standardTransfer
├── TRIM
└── zippedDirectory
```

File: src/dashboard/frontend/app/services/transfer_browser_transfer.service.js [hunk 4]
```diff
@@ -46,7 +46,13 @@ class TransferBrowserTransfer {
   start() {
     // If this is a zipped bag, then there will be no transfer name;
     // give it a dummy name instead.
-    let name = this.type === 'zipped bag' ? 'ZippedBag' : this.name;
+    let name = this.name;
+    if (this.type === 'zipped bag') {
+      name = 'ZippedBag';
+    } else if (this.type == 'zipfile') {
```
Comment (ross-spencer) [inline src/dashboard/frontend/app/services/transfer_browser_transfer.service.js]: It looks like this should be the `===` strict comparison as well?

**Review by ross-spencer — CHANGES_REQUESTED** [review]
Hi @alexwlchan I have reviewed this and it looks good. Please review the CR notes and let us know the capacity you have to address the rebase this week. While Helen and I were coordinating other PRs, we lost track of the need to rebase this one following changes we made to preserve process configuration in zipped transfers/ You'll see in my notes I submitted what I think is a merged version of [`extract_zipped_transfer.py`](https://github.com/artefactual/archivematica/pull/1556/commits/8ad107d5e6c2c41644e4825a1023bc7dfb8da33f). That needs checking alongside what you're running, and perhaps testing alongside what you desire to be running. Once that's merged, the rebase to `qa/1.x` looks straightforward.

This was a fascinating one to review. I learned a little more about the capability of zip-like transfers and found two `qa/1.x` regressions **nb.** not caused by this PR.

You'll see in the primary body of the PR submitted by Helen I have suggested some docs [here](https://github.com/artefactual/archivematica/pull/1556). It's a very light treatment. Please feel free to review also.

**Comment by alexwlchan** [comment]
@ross-spencer I think I’ve addressed the review markups, fancy another look?

**Comment by alexwlchan** [comment]
why travis 😭 

(investigating)

File: src/MCPClient/lib/clientScripts/extract_zipped_transfer.py [hunk 5]
```diff
@@ -87,20 +87,9 @@ def call(jobs):
...
-                try:
-                    tar_extension_position = destinationDirectory.rindex(".tar")
-                    destinationDirectory = destinationDirectory[:tar_extension_position]
-                except ValueError:
-                    pass
-
-                zipLocation = os.path.join(
-                    processingDirectory, os.path.basename(target)
-                )
+                destinationDirectory = get_extract_dir_name(destinationDirectory)
```
Comment (ross-spencer) [inline src/MCPClient/lib/clientScripts/extract_zipped_transfer.py]: Looking good, we just need to make sure `get_extract_dir_name(..)` is imported at the top of the file:

* `from fileOperations import get_extract_dir_name`

File: src/MCPClient/lib/clientScripts/extract_zipped_transfer.py [hunk 6]
```diff
@@ -87,20 +87,9 @@ def call(jobs):
...
-                # trim off '.tar' if present (os.path.basename doesn't deal well with '.tar.gz')
-                try:
-                    tar_extension_position = destinationDirectory.rindex(".tar")
-                    destinationDirectory = destinationDirectory[:tar_extension_position]
-                except ValueError:
-                    pass
-
-                zipLocation = os.path.join(
-                    processingDirectory, os.path.basename(target)
-                )
```
Comment (ross-spencer) [inline src/MCPClient/lib/clientScripts/extract_zipped_transfer.py]: The build tells me 101 to 103 need to be reinstated :)

**Review by ross-spencer — CHANGES_REQUESTED** [review]
Thanks for the quick response @alexwlchan just two changes that I can see, it looks like they were highlighted by the linter too. I think those changes should make the build happy. If you wanted to give rebasing to apply these changes on top of Artefactual's `qa/1.x` as well to be extra sure about the build, that might be a good idea, but I think when we squish that all will be taken care off anyway.

**Comment by alexwlchan** [comment]
🤦‍♀ 

Pushed a fixed commit.

**Comment by ross-spencer** [comment]
Perfect, thank you @helenst and @alexwlchan for this! Will get this merged and so it should appear in `1.11` soon!! 

image

**Review by ross-spencer — APPROVED** [review]
Perfect, thank you @helenst and @alexwlchan for this! Will get this merged and so it should appear in `1.11` soon!! 

image

**Pull request merged**